snuba/datasets/schema.py

```diff
@@ -75,11 +70,59 @@ def get_column_differences(self, expected_columns: Mapping[str, str]) -> List[st
 
     def get_migration_statements(
         self
-    ) -> Callable[[str, Mapping[str, str]], Sequence[str]]:
+    ) -> Optional[Callable[[str, Mapping[str, str]], Sequence[str]]]:
         return self.__migration_function
 
 
 class TableSchema(Schema):
+    """
+    Represent a table-like schema. This means it represents either
+    a Clickhouse table, a Clickhouse view or a Materialized view.
```

Comment:
Is this supposed to be explicitly writable? Or is that not a requirement here yet?

If so, should it be called `WritableTableSchema` or similar?

Reply:
As of now it does not (the materialized view extends this). 
But I think this is conflating two concerns: the table abstraction with DDL (thus access to table name) and the writable abstraction which means returned by DatasetSchemas.get_write_schema.

I think we can get away with making WritableTable extend Table and MaterializedView extend Table.
My concern is that being a table and being writable should be orthogonal concerns but I cannot find a concrete example we would run into troubles with by making WritableTable extend Table.

Will add the class in between for now